CMakeLists.txt

```diff
@@ -288,6 +289,7 @@ target_compile_definitions(celerity_runtime PUBLIC
   CELERITY_FEATURE_UNNAMED_KERNELS=$<BOOL:${CELERITY_FEATURE_UNNAMED_KERNELS}>
   CELERITY_DETAIL_HAS_NAMED_THREADS=$<BOOL:${CELERITY_DETAIL_HAS_NAMED_THREADS}>
   CELERITY_ACCESSOR_BOUNDARY_CHECK=$<BOOL:${CELERITY_ACCESSOR_BOUNDARY_CHECK}>
+  CELERITY_ACCESS_PATTERN_DIAGNOSTICS=$<BOOL:${CELERITY_ACCESS_PATTERN_DIAGNOSTICS}>
```

Comment:
Adding this line made me realize two things:

1. We need to generate a `config.h`, the number of compiler flags is getting out of hand.
2. We're not writing the values of `CELERITY_DETAIL_HAS_NAMED_THREADS` or `CELERITY_ACCESSOR_BOUNDARY_CHECK` to the generated CMake config (`celerity-config.cmake.in`). Is this an oversight or correct behavior?

Reply:
1. Yes.
2. For the latter I would assume it's an oversight; I guess the former is not really relevant / actionable to users (also it's in the `DETAIL` namespace).

Reply:
I opened an internal issue for 1) and added all three flags to `cmake.in`.